packages/@tinacms/graphql/src/database/index.ts

```diff
@@ -1000,16 +1020,46 @@ export class Database {
         await level.batch(batchOps)
       }
     }
+    // This map is used to map files to there collections
+    const filesSeen = new Map<string, string[]>()
+    // This is used to track which files have duplicate collections so we do not have to loop over all files at the end
+    const duplicateFiles = new Set<string>()
     await sequential(tinaSchema.getCollections(), async (collection) => {
-      const documentPaths = await this.bridge.glob(
-        normalizePath(collection.path),
-        collection.format || 'md'
-      )
-      await _indexContent(this, level, documentPaths, enqueueOps, collection)
+      const normalPath = normalizePath(collection.path)
+      const format = collection.format || 'md'
+      // Get all possible paths for this collection
+      const documentPaths = await this.bridge.glob(normalPath, format)
+      // filter paths based on match
+      const match = `${normalPath}/${collection.match}.${format}`
+      const filteredPaths = collection.match
+        ? micromatch(documentPaths, [match])
+        : documentPaths
+
+      filteredPaths.forEach((path) => {
+        if (filesSeen.has(path)) {
+          filesSeen.get(path).push(collection.name)
+          duplicateFiles.add(path)
+        } else {
+          filesSeen.set(path, [collection.name])
+        }
+      })
+      duplicateFiles.forEach((path) => {
+        warnings.push(
+          // TODO: link to docs
+          `"${path}" Found in collections multiple collections: ${filesSeen
+            .get(path)
+            .join(
+              ', '
+            )}. This can cause unexpected behavior. We recommend updating the \`match\` property of those collections so that each file is in only one collection.`
+        )
+      })
+
+      await _indexContent(this, level, filteredPaths, enqueueOps, collection)
     })
     while (operations.length) {
       await level.batch(operations.splice(0, 25))
     }
+    return { warnings }
```

Comment:
should we have a return type on the functions in this that return the warnings? Also wondering if it would be better to error out rather than emit warnings? @jamespohalloran or @jeffsee55 have thoughts on that?

Reply:
Hmm, if the behaviour's wonky for overlapping collections, an error does seem appropriate to me.
It is tricky the amount of overlap we'll likely face for Forestry sites (especially if we're going to use the `match` field to handle single document collections 🤔)

Reply:
Only reason I am leaning towards warnings is to remain backwards compatible

Reply:
Also, I _think_ the return type is inferred.

Reply:
what do you mean by backwards compatible? are there sites using `match` currently?